src/index.js

```diff
@@ -42,49 +38,39 @@ function flatten(node) {
   parts.unshift(name);
 
   return { name, keypath: parts.join(".") };
-}
+};
 
 export default function inject(options) {
   if (!options) throw new Error("Missing options");
 
   const filter = createFilter(options.include, options.exclude);
 
-  let modules;
+  let modules = options.modules;
 
-  if (options.modules) {
-    modules = options.modules;
-  } else {
+  if (!modules) {
     modules = Object.assign({}, options);
     delete modules.include;
     delete modules.exclude;
   }
 
+  const modulesMap = new Map(Object.entries(modules));
+
   // Fix paths on Windows
   if (sep !== "/") {
-    Object.keys(modules).forEach(key => {
-      const module = modules[key];
-
-      modules[key] = Array.isArray(module)
-        ? [module[0].split(sep).join("/"), module[1]]
-        : module.split(sep).join("/");
+    modulesMap.forEach((mod, key) => {
+      modulesMap.set(key, Array.isArray(mod)
+        ? [mod[0].split(sep).join("/"), mod[1]]
+        : mod.split(sep).join("/"));
     });
   }
 
-  const firstpass = new RegExp(
-    `(?:${Object.keys(modules)
-      .map(escape)
-      .join("|")})`,
-    "g"
-  );
```

Comment:
I do not think it will be a sure performance win to skip the first pass. The AST will ONLY be reused if this is the last plugin doing a `transform`, which is easily not the case if people use many plugins, which again is often the case for larger code-bases. I would rather leave it.